R/intervals_support_funs.R

```diff
@@ -0,0 +1,295 @@
+      impute_methods <- unlist(strsplit(ifelse(is.na(.data[[impute_col]]), "", .data[[impute_col]]), ","))
+      if (!allow_duplication && target_impute %in% impute_methods) {
+        # If duplication is not allowed, do not add the impute method if it already exists
+        .data[[impute_col]]
+      } else {
+        # Add the impute method after the specified position
+        impute_methods <- append(impute_methods, target_impute, after)
+        paste(impute_methods[impute_methods != ""], collapse = ",")
+      }
+    })) %>%
+    dplyr::ungroup() %>%
+    dplyr::mutate(index = if (new_rows_after_original) index + 0.5 else index + max(index)) %>%
+    as.data.frame()
+  
+  # Eliminate from the old intervals the target parameters
+  old_intervals_without_impute <- target_intervals %>%
+    dplyr::mutate(dplyr::across(dplyr::any_of(target_params), ~FALSE))
+  
+  # Make parameters FALSE in original intervals and join the new ones
+  data <- data %>%
+    dplyr::anti_join(target_intervals, by = names(data)) %>%
+    dplyr::bind_rows(old_intervals_without_impute, new_intervals_with_impute) %>%
+    dplyr::filter(rowSums(dplyr::across(dplyr::any_of(param_cols), as.numeric)) > 0) %>%
+    dplyr::arrange(index) %>%
+    dplyr::select(-index)
+  
+  data
+}
+
+
+#' Remove specified imputation methods from the intervals in a PKNCAdata or data.frame object.
+#'
+#' @inheritParams interval_add_impute
+#' @param target_impute A character string specifying the imputation method to be removed.
+#' @return A modified object with the specified imputation methods removed from the targeted intervals.
+#' @examples
+#' d_conc <- data.frame(
+#'   conc = c(1, 0.6, 0.2, 0.1, 0.9, 0.4, 1.2, 0.8, 0.3, 0.2, 1.1, 0.5),
+#'   time = rep(0:5, 2),
+#'   ID = rep(1:2, each = 6),
+#'   analyte = rep(c("Analyte1", "Analyte2"), each = 6),
+#'   include_hl = c(FALSE, NA, TRUE, TRUE, TRUE, TRUE, FALSE, NA, TRUE, TRUE, TRUE, TRUE)
+#' )
+#'
+#' d_dose <- data.frame(
+#'   dose = c(100, 200),
+#'   time = c(0, 0),
+#'   ID = c(1, 2)
+#' )
+#'
+#' o_conc <- PKNCAconc(d_conc, conc ~ time | ID / analyte)
+#' o_dose <- PKNCAdose(d_dose, dose ~ time | ID)
+#'
+#' intervals <- data.frame(
+#'   start = c(0, 0, 0),
+#'   end = c(3, 5, Inf),
+#'   half.life = c(TRUE, FALSE, TRUE),
+#'   cmax = c(TRUE, TRUE, TRUE),
+#'   impute = c("start_conc0,start_predose", "start_predose", "start_conc0"),
+#'   analyte = c("Analyte1", "Analyte2", "Analyte1")
+#' )
+#'
+#' o_data <- PKNCAdata(o_conc, o_dose, intervals = intervals)
+#'
+#' # Apply interval_remove_impute function
+#' o_data <- interval_remove_impute(data = o_data, target_impute = "start_conc0", target_params = c("half.life"), target_groups = data.frame(analyte = "Analyte1"))
+#' @export
+interval_remove_impute <- function(data, target_impute, target_params = NULL, target_groups = NULL, impute_column = NULL, new_rows_after_original = TRUE) {
+  if (missing(data) || missing(target_impute)) {
```

Comment:
Please apply the same comments as `interval_add_impute` for the generic method here.